Enhance Filters (Invoke-FGTRestMethod)

Tuurtje opened the pull request:
There was something wrong with the parameter sets. Somehow they need to be on the same line as "mandatory", else all options are visible in every parameterset. 

Also added the remaining filter type's (because i needed them).

alagoutte commented:
Hi Arthur,

Thanks, All Filter work ?!

Yes, it is strange for filter... (i really need to add some test...)

Tuurtje commented:
Yeah tested them all. they all seem to work.

________________________________
Van: Alexis La Goutte <notifications@github.com>
Verzonden: vrijdag 10 april 2020 09:25
Aan: FortiPower/PowerFGT <PowerFGT@noreply.github.com>
CC: Arthur Heijnen <arthur.heijnen@live.nl>; Author <author@noreply.github.com>
Onderwerp: Re: [FortiPower/PowerFGT] Filters (#95)


Hi Arthur,

Thanks, All Filter work ?!

Yes, it is strange for filter... (i really need to add some test...)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/FortiPower/PowerFGT/pull/95#issuecomment-611956194>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABQAF76KF46QF4JHTCVP5X3RL3QXXANCNFSM4MFJYI7A>.

alagoutte commented:
Thanks, i will check !

alagoutte commented on PowerFGT/Public/cmdb/firewall/address.ps1:
```diff
@@ -209,17 +219,16 @@ function Get-FGTFirewallAddress {
-        [Parameter (Mandatory = $false)]
-        [Parameter (ParameterSetName = "filter")]
+        [Parameter(Mandatory = $false, ParameterSetName = "filter")]
+        [String]$filter,
```
i don't really like the idea to have directly filter parameter.. but prefer we can build filter with filter_type, attribute...

Tuurtje replied:
Oke i will remove it. Advanced searches can be done with invoke-FGTRestMothode if needed. is the parameter name-change  (filter -> filter_build) oke?  This will make it consistant with Invoke-FGTRestMethode. If not i will change it back.

alagoutte replied:
Ok for change ParameterName

alagoutte commented:
Thanks Arthur, can you add name of file on commit message ?
ippool: Fixed ParameterSets and added filter types 

for example (or make a big commit with all change)
it is very strange, i don't work if you set the parametername outside of first Parameter

Tuurtje commented:
Will do next time. I made the same change in all file. Don't know how to do this in one commit.

________________________________
Van: Alexis La Goutte <notifications@github.com>
Verzonden: dinsdag 14 april 2020 08:16
Aan: FortiPower/PowerFGT <PowerFGT@noreply.github.com>
CC: Arthur Heijnen <arthur.heijnen@live.nl>; Author <author@noreply.github.com>
Onderwerp: Re: [FortiPower/PowerFGT] Enhance Filters (Invoke-FGTRestMethod) (#95)


Thanks Arthur, can you add name of file on commit message ?
ippool: Fixed ParameterSets and added filter types

for example (or make a big commit with all change)
it is very strange, i don't work if you set the parametername outside of first Parameter

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/FortiPower/PowerFGT/pull/95#issuecomment-613294845>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABQAF77EKVSLJIWBZW4KOJDRMQLVFANCNFSM4MFJYI7A>.

alagoutte commented:
What tools do you are using for git ?

only need to make a git add of all file (or use git commit -a) after make change all files

(you can also use git rebase -i master and reword/squash/fixup and after need to make a git push -f for force)

Tuurtje commented:
Copy past in webbrowser :-( I've got github desktop installed. will check the howto.

________________________________
Van: Alexis La Goutte <notifications@github.com>
Verzonden: dinsdag 14 april 2020 08:56
Aan: FortiPower/PowerFGT <PowerFGT@noreply.github.com>
CC: Arthur Heijnen <arthur.heijnen@live.nl>; Author <author@noreply.github.com>
Onderwerp: Re: [FortiPower/PowerFGT] Enhance Filters (Invoke-FGTRestMethod) (#95)


What tools do you are using for git ?

only need to make a git add of all file (or use git commit -a) after make change all files

(you can also use git rebase -i master and reword/squash/fixup and after need to make a git push -f for force)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/FortiPower/PowerFGT/pull/95#issuecomment-613314910>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABQAF743VQ3NGDKMFAVPVSDRMQQNRANCNFSM4MFJYI7A>.

poundy commented:
> Copy past in webbrowser :-( I've got github desktop installed. will check the howto.
> […](#)
Welcome to that club :) 
I'm waiting for the 0.5.0 release so I can get back to a state to test and tweak further because I started with copy/paste and not understanding pull requests :)

alagoutte commented:
Hi sorry guys but busy for the moment for work on PowerFGT (need to finish 2 other module)
I hope get more time end of month and release a new version next month (i will add all external contribution and add tests integration)

I can help you, if you need for git skill

alagoutte commented:
Hi Arthur 

you can kept open it is always on my todo (but no really in the top !)
